### PR TITLE
improve card response usage + formatting

### PR DESCRIPTION
**Pull request by miasya**
- add Url to cardResponse.
- remove keyvalue widget. 
- add logic to Bot.java that will return a text message or card depending on what's more appropriate

### CODE REVIEW

File: src/main/java/com/google/flourbot/api/Bot.java [hunk 1]
```diff
@@ -79,9 +80,17 @@ public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecu
         String threadId = event.at("/message/thread/name").asText();
         
         ChatResponse chatResponse = macroExecutionModule.execute(email, message, threadId);
+        
         replyText = chatResponse.getReplyText();
-        Card card = CardResponse.createCardResponse(replyText);
-        reply.setCards(Collections.singletonList(card));
+        ActionType actionType = chatResponse.getActionType();
+        String documentUrl = chatResponse.getDocumentUrl();
```
Comment (tsa87) [inline src/main/java/com/google/flourbot/api/Bot.java:89]: What is `documentUrl` here?

**Review by tsa87 — APPROVED** [review]
LGTM!

File: src/main/java/com/google/flourbot/api/Bot.java (hunk 1 above)
Comment (miasya) [reply src/main/java/com/google/flourbot/api/Bot.java:89]: sheetUrl so it creates a button to open up the doc in browser basically

File: src/main/java/com/google/flourbot/api/Bot.java [hunk 2]
```diff
@@ -79,9 +80,17 @@ public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecu
...
+
+        // Create card if appropriate
+        if (actionType != null && documentUrl != null) {
```
Comment (SydneyCaulfeild) [inline src/main/java/com/google/flourbot/api/Bot.java:92]: I like having the logic to only return a card when we need one! I had cards that just returned text for the Resource Helper bot but it looked kind of strange.

File: src/main/java/com/google/flourbot/api/DriveCloudSheet.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
 public class DriveCloudSheet implements CloudSheet {
   private static final String VALUE_INPUT_OPTION = "USER_ENTERED";
   private static final String INSERT_DATA_OPTION = "INSERT_ROWS";
+  private static final String SHEET_NAME = "Sheet1"; // Stretch goal: allow user specified
```
Comment (SydneyCaulfeild) [inline src/main/java/com/google/flourbot/api/DriveCloudSheet.java:22]: Does this automatically name the sheet they provided in the website to Sheet1?

**Review by SydneyCaulfeild — APPROVED** [review]
I really like the card it returns where it shows a screenshot of the current spreadsheet, good work!

File: src/main/java/com/google/flourbot/api/Bot.java (hunk 1 above)
Comment (tsa87) [reply src/main/java/com/google/flourbot/api/Bot.java:89]: Oh okay neat!

File: src/main/java/com/google/flourbot/api/Bot.java (hunk 2 above)
Comment (lipczak-g) [reply src/main/java/com/google/flourbot/api/Bot.java:92]: +1 maybe we can have a constant list of action types for which we display cards? Anything that generates some form of structured information, like a list of values from a spreadsheet.

The link to the spreadsheet is a great feature, but I think it should be controlled by the macro creator, sometimes you may not want to disclose to the users that the data is stored in a specific file.

File: src/main/java/com/google/flourbot/api/DriveCloudSheet.java (hunk 3 above)
Comment (miasya) [reply src/main/java/com/google/flourbot/api/DriveCloudSheet.java:22]: Naw, it just assumes they created the sheet and didn't rename it away from the default

**Pull request merged**